### PR TITLE
Add query language functions to get data structure sizes.

### LINKED ISSUE #2883: Add a method to the query library to get data structure sizes
When using the query library and the query language, it is very common to need to get the size of a `Vector` or of a Java array.  Unfortunately, `Vector` uses `.size()`, and Java arrays use `.length`.  This makes code nastier than needed.  Consider adding methods to the query library to make this easier on users.  Also consider using the python `len` naming convention.

### PR DESCRIPTION
**Pull request by chipkent**
Add query language functions to get data structure (`Vector` and java array) sizes.

Fixes #2883.

### CODE REVIEW

File: engine/function/src/templates/Basic.ftl [hunk 1]
```diff
@@ -131,6 +132,35 @@ public class Basic {
...
+
+    /**
+     * Returns the length of the input.
+     *
+     * @param values values.
+     * @return length of the input or zero for null inputs.
+     */
+    @SuppressWarnings("rawtypes")
+    static public long len(LongSizedDataStructure values) {
+        if(values == null){
```
Comment (kosak) [inline engine/function/src/templates/Basic.ftl:157]: space after "if" and also space before opening brace :-)

**Review by kosak — APPROVED** [review]
Two whitespace nits, but approved.

File: engine/function/src/templates/Basic.ftl [hunk 2]
```diff
@@ -131,6 +132,35 @@ public class Basic {
...
+
+    /**
+     * Returns the length of the input.
+     *
+     * @param values values.
+     * @return length of the input or the Deephaven null constant for null inputs.
+     */
+    @SuppressWarnings("rawtypes")
+    static public long len(LongSizedDataStructure values) {
+        if (values == null){
```
Comment (kosak) [inline engine/function/src/templates/Basic.ftl:157]: one more space (between closing parenthesis and open brace) and I'll be happy :-)

**Review by kosak — APPROVED** [review]
Approved with one whitespace nit

**Review by kosak — APPROVED** [review]

**Comment by niloc132** [comment]
Ping @nbauernfeind who suggested instead (or "also") supporting .size() on arrays, so that the code is not just consistent with python, but consistent with java (the syntax used in these kinds of expressions).

**Pull request merged**